### PR TITLE
Better docs: Add usage section to README + add comments clarifying limits on context/player

### PR DESCRIPTION
**Pull request by bloeys**
Based on discussion about better docs [here](https://github.com/hajimehoshi/oto/issues/136), I am submitting this PR to add a new 'Usage' section to the README with code examples, and comments on some structs/functions of the context and player to clarify when its safe to create multiple and when not.

The usage section also includes some examples of sync-playing audio and in-memory/streaming files. It would be great if someone more experienced in Oto can double check that logic and confirm its good.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,6 +57,150 @@ In most cases this command must be run by root user or through `sudo` command.
...
 
+## Usage
+
+The two main components of Oto are a `Context` and `Players`. The context handles interactions with
+the OS and audio drivers, and as such there can only be **one** context in your program.
+
+From a context you can create any number of different players, where each player is given an `io.Reader` that
+it reads bytes representing sounds from and plays.
+
+> Note that a single `io.Reader` must **not** be used by multiple players.
```
Comment (hajimehoshi) [inline README.md:68]: Is this a quote from somewhere?

File: README.md [hunk 2]
```diff
@@ -44,6 +57,150 @@ In most cases this command must be run by root user or through `sudo` command.
...
+import (
+    "github.com/hajimehoshi/go-mp3"
+    "github.com/hajimehoshi/oto/v2"
+    "time"
+    "os"
+)
+
+func main() {
+
+    //Read the mp3 file into memory
```
Comment (hajimehoshi) [inline README.md:86]: Add space after `//`

File: README.md [hunk 3]
```diff
@@ -44,6 +57,150 @@ In most cases this command must be run by root user or through `sudo` command.
...
+    fileBytes, err := os.ReadFile("./my-file.mp3")
```
Comment (hajimehoshi) [inline README.md:87]: os.Open? Or, the binary cannot be passed to mp3.NewDecoder, and actually it is not passed. Or do you mean using byte.Reader?

File: README.md [hunk 4]
```diff
@@ -44,6 +57,150 @@ In most cases this command must be run by root user or through `sudo` command.
...
+doesn't mean they were all played from the audio device.
+
+Data is moved from io.Reader->internal buffer->audio device, and when the internal buffer moves data to the audio device
+is not guaranteed, so there might be a small delay. The amount of data in the buffer can be retrieved
+using `Player.UnplayedBufferSize()`.
+
+The size of the underlying buffer of a player can also be set by type-asserting the player object:
+
+```go
+myPlayer.(BufferSizeSetter).SetBufferSize(newBufferSize)
```
Comment (hajimehoshi) [inline README.md:199]: `myPlayer.(oto.BufferSizeSetter).SetBufferSize(newBufferSize)`

**Review by hajimehoshi — COMMENTED** [review]
Thank you!

File: README.md (hunk 3 above)
Comment (bloeys) [reply README.md:87]: Oh good catch!
Yes I meant to use a `bytes.NewReader`, which is now fixed.

File: README.md (hunk 1 above)
Comment (bloeys) [reply README.md]: No but I like to use it sometimes to emphasis things.
Removed.

File: README.md (hunk 2 above)
Comment (bloeys) [reply README.md]: Added space to all comments.

File: README.md (hunk 4 above)
Comment (bloeys) [reply README.md]: Done, thanks :)

File: README.md [hunk 5]
```diff
@@ -44,6 +57,153 @@ In most cases this command must be run by root user or through `sudo` command.
...
+
+import (
+    "github.com/hajimehoshi/go-mp3"
+    "github.com/hajimehoshi/oto/v2"
+    "time"
+    "os"
+)
+
+func main() {
+
```
Comment (hajimehoshi) [inline README.md:85]: Remove this empty line

File: README.md [hunk 6]
```diff
@@ -44,6 +57,153 @@ In most cases this command must be run by root user or through `sudo` command.
...
+The following is an example of loading and playing an MP3 file:
+
+```go
+package main
+
+import (
+    "github.com/hajimehoshi/go-mp3"
+    "github.com/hajimehoshi/oto/v2"
+    "time"
+    "os"
```
Comment (hajimehoshi) [inline README.md:81]:
```
    "time"
    "os"

    "github.com/hajimehoshi/go-mp3"
    "github.com/hajimehoshi/oto/v2"
```

File: README.md [hunk 7]
```diff
@@ -44,6 +57,153 @@ In most cases this command must be run by root user or through `sudo` command.
...
+
+func main() {
+
+    // Read the mp3 file into memory
+    fileBytes, err := os.ReadFile("./my-file.mp3")
+    if err != nil {
+        panic("Failed to load mp3")
+    }
+
+    //Convert the pure bytes into a reader object that can be used with the mp3 decoder
```
Comment (hajimehoshi) [inline README.md:92]: Add space after //

File: README.md [hunk 8]
```diff
@@ -44,6 +57,153 @@ In most cases this command must be run by root user or through `sudo` command.
...
+    fileBytesReader := bytes.NewReader(fileBytes)
+
+    // Decode file
+    decodedMp3, err := mp3.NewDecoder(fileBytesReader)
+    if err != nil{
```
Comment (hajimehoshi) [inline README.md:97]: add space after nil

File: README.md [hunk 9]
```diff
@@ -44,6 +57,153 @@ In most cases this command must be run by root user or through `sudo` command.
...
+In such cases you might want to stream the file. Luckily this is very simple, just use `os.Open`:
+
+```go
+package main
+
+import (
+    "github.com/hajimehoshi/go-mp3"
+    "github.com/hajimehoshi/oto/v2"
+    "time"
+    "os"
```
Comment (hajimehoshi) [inline README.md:159]: ditto

File: README.md [hunk 10]
```diff
@@ -44,6 +57,153 @@ In most cases this command must be run by root user or through `sudo` command.
...
+)
+
+func main() {
+
```
Comment (hajimehoshi) [inline README.md:163]: ditto

File: README.md [hunk 11]
```diff
@@ -44,6 +57,153 @@ In most cases this command must be run by root user or through `sudo` command.
...
+    player.Play()
+
+    // We can wait for the sound to finish playing using something like this
+    for player.IsPlaying() {
+        time.Sleep(time.Millisecond)
+    }
+
+    // Now that the sound finished playing, we can restart from the beginning with these two lines
+    player.Reset()
+    player.Play()
```
Comment (hajimehoshi) [inline README.md:135]: So, in this example, this play immediately stops, right? I think this seems confusing. What about removing this Reset and Play?

File: README.md (hunk 5 above)
Comment (bloeys) [reply README.md:85]: Done

File: README.md (hunk 11 above)
Comment (bloeys) [reply README.md:135]: Well I wanted to show how to rewind/replay. Maybe we can put the loop here again or a simple time.Sleep?

File: README.md (hunk 6 above)
Comment (bloeys) [reply README.md:79]: Done :)

File: README.md (hunk 7 above)
Comment (bloeys) [reply README.md]: Done

File: README.md (hunk 8 above)
Comment (bloeys) [reply README.md]: Added

File: README.md (hunk 10 above)
Comment (bloeys) [reply README.md]: dittoed

File: README.md (hunk 11 above)
Comment (hajimehoshi) [reply README.md:135]: Or just commenting out these line would be fine.

Comment (bloeys) [reply README.md]: Commented out 👍🏻

**Review by hajimehoshi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**